src/pages/list.tsx

```diff
@@ -14,16 +14,14 @@ const List = () => {
   const router = useRouter();
   const { search } = router.query;
   const [page, setPage] = useState(1);
-  const {
-    data: beersData,
-    fetchNextPage,
-    isLoading,
-  } = useAllBeers({
+  const { data: beersData, fetchNextPage, isLoading } = useAllBeers({
     page,
     per_page: LIST_PER_PAGE,
     isPreferenceOrRateChecked: true,
   });
 
+  const { data: searchList } = useSearchBeer({ name: `${search}` });
+
   const beerList = useMemo(() => flatten(beersData?.pages?.map(page => page) ?? []), [beersData]);
 
```

Comment:
beerList 랑 searchList랑 하나의 변수로 해서 `search`가 있으면 searchList로 없으면 beerList 반환하는건 어떨까요?

```suggestion
  const { data: searchList } = useSearchBeer({ name: `${search}` });
  
  const beerList = useMemo(() => flatten(beersData?.pages?.map(page => page) ?? []), [beersData]);
  const beers = useMemo(()=> search? searchList: beerList,[beerList,searchList,search])
```

Reply:
넵 죠습니다!